kernel/src/transaction/mod.rs

```diff
@@ -159,7 +161,8 @@ impl Transaction {
         let commit_timestamp = current_time_ms()?;
 
         Ok(Transaction {
-            read_snapshot,
+            read_snapshot: read_snapshot.clone(),
+            committer: Box::new(FileSystemCommitter::new(read_snapshot)),
```

Comment:
This seems like a sharp edge. I don't think we should default to a file system committer if we just give a snapshot, since that's not the semantics that just having a snapshot imply. Could we instead take the committer as an argument so callers have to be explicit?

Reading lower I see you do catch the case that we passed a cc table here, but I still prefer explicit at compile time over a runtime error.

Reply:
Hm yea fair - I'm mostly attempting to keep the `FileSystemCommitter` an internal detail though - for non-cc tables everything can just proceed as normal, but for cc tables we check that you did pass in a different committer.

What do you think? Do you think it's better to just expose `FileSystemCommitter` as `pub` and then make all callers include a `Committer` explicitly in the builder? I do agree that perhaps being explicit is nice but it comes at the cost of users having to suddenly grok a "filesystem committer"..